server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -169,8 +169,8 @@ public void setProfiler(QueryProfiler profiler) {
      * Add a {@link Runnable} that will be run on a regular basis while accessing documents in the
      * DirectoryReader but also while collecting them and check for query cancellation or timeout.
      */
-    public Runnable addQueryCancellation(Runnable action) {
-        return this.cancellable.add(action);
+    public void addQueryCancellation(Runnable action) {
```

Comment:
Is it relevant to the pr, or is it just a code update related to the functionality?

Reply:
good catch. it's somewhat related as it's another cleanup in the same area of the code - maybe it is not super evident that search timeout is implemented via the query cancellation mechanism. I could have opened a separate PR for this change.